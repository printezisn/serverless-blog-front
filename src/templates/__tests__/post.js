import React from "react"
import renderer from "react-test-renderer"

import Post from "../post"

jest.mock("../../components/layout", () => "Layout")
jest.mock("../../components/markdown", () => "Markdown")

describe("Blog post page", () => {
    it("renders correctly", () => {
        const post = {
            id: "test-id",
            title: "test-title",
            description: "test-description",
            body: "test-body",
        }
        const pageContext = { post }
        const template = renderer.create(<Post pageContext={pageContext} />)

        expect(template.toJSON()).toMatchSnapshot()
    })
})
