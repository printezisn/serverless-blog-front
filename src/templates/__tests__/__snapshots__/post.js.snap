// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog post page renders correctly 1`] = `
<Layout>
  <article>
    <header>
      <h2
        className="title is-4 has-text-centered"
      >
        test-title
      </h2>
    </header>
    <section
      className="blog-post"
    >
      <Markdown
        text="test-body"
      />
    </section>
  </article>
</Layout>
`;
